contrib/python-bindings/include/point_wrapper.h

```diff
@@ -51,12 +56,89 @@ namespace python
     ~PointWrapper();
 
     /**
+     * Return the Euclidean distance of this point to the point p, i.e., the
+     * l2_norm of the difference between the vectors representing the two
+     * points.
+     */
+    double distance(const PointWrapper &p) const;
+
+    /**
+     * Return the l2_norm of the vector.
+     */
+    double norm() const;
+
+    /**
+     * Return the sum of the absolute squares of all entries.
+     */
+    double norm_square() const;
+
+    /**
      * Assignment operator. The dimension of the point is changed if it is
      * different than the one of @p other.
      */
     PointWrapper &operator= (const PointWrapper &other);
 
     /**
+     * Test for inequality of two points.
+     */
+    bool operator!= (const PointWrapper &p) const; 
+
+    /**
+     * Test for equality of two tensors.
+     */
+    bool operator== (const PointWrapper &p) const;
+
+    /**
+     * Return the scalar product of the vectors representing two points.
```

Comment:
Yes, we could add a TensorWrapper but unless we want to have a complete python interface, I don't see the point of adding it.